src/MAUI/Maui.Samples/Samples/Layers/AddVectorTiledLayerFromCustomStyle/AddVectorTiledLayerFromCustomStyle.xaml.cs

```diff
@@ -0,0 +1,140 @@
+            };
+
+            // Load the default portal.
+            ArcGISPortal portal = await ArcGISPortal.CreateAsync();
+
+            // Create vector tiled layers for each portal item.
+            _vectorTiledLayers = new List<ArcGISVectorTiledLayer>();
+
+            // Store a list of tiled layers in the different styles.
+            foreach (string item in _itemIDs)
+            {
+                PortalItem websceneItem = await PortalItem.CreateAsync(portal, item);
+                ArcGISVectorTiledLayer vectorTiledLayer = new ArcGISVectorTiledLayer(websceneItem);
+                _vectorTiledLayers.Add(vectorTiledLayer);
+            }
+
+            // Create a map with the default style.
+            MyMapView.Map = new Map(SpatialReferences.WebMercator);
+            MyMapView.Map.Basemap.BaseLayers.Add(_vectorTiledLayers[0]);
+
+            await MyMapView.SetViewpointAsync(_defaultViewpoint);
+        }
+
+        private async void Export(ArcGISVectorTiledLayer vectorTiledLayer)
+        {
+            // Create the task.
+            ExportVectorTilesTask exportTask = await ExportVectorTilesTask.CreateAsync(vectorTiledLayer.Source);
+
+            ExportVectorTilesParameters parameters = await exportTask.CreateDefaultExportVectorTilesParametersAsync(MyMapView.GetCurrentViewpoint(ViewpointType.BoundingGeometry).TargetGeometry, MyMapView.MapScale * 0.1);
+
+            // By using the UseReducedFontService download option the file download speed is reduced.
+            // This limits vector tiled layers character sets and may not be suitable in every use case.
+            parameters.EsriVectorTilesDownloadOption = EsriVectorTilesDownloadOption.UseReducedFontsService;
+
+            // Get the tile cache path and item resource path for the base layer styling.
+            string tilePath = Path.Combine(Environment.ExpandEnvironmentVariables("%TEMP%"), Path.GetTempFileName() + ".vtpk");
+            string itemResourcePath = Path.Combine(Environment.ExpandEnvironmentVariables("%TEMP%"), Path.GetTempFileName() + "_styleItemResources");
+
+            // Create the export job and start it.
+            ExportVectorTilesJob _job = exportTask.ExportVectorTiles(parameters, tilePath, itemResourcePath);
+            _job.Start();
+
+            // Wait for the job to complete.
+            ExportVectorTilesResult vectorTilesResult = await _job.GetResultAsync();
+
+            // Load the vector tile cache.
+            VectorTileCache vectorTileCache = vectorTilesResult.VectorTileCache;
+            await vectorTileCache.LoadAsync();
+        }
+
+        private async void StyleChooserClick(object sender, EventArgs e)
+        {
+            try
+            {
+                // Get the chosen style and determine that style's index.
+                string style = await Application.Current.MainPage.DisplayActionSheet("Select:", "Cancel", null, styles);
+                int styleIndex = Array.IndexOf(styles, style);
```

Comment:
```suggestion
                if(style.Equals("Cancel")) return;
                int styleIndex = Array.IndexOf(styles, style);
```
Instead of catching the exception, you can handle the Cancel directly.